src/sequence_statistics.py

```diff
@@ -0,0 +1,229 @@
+import requests
+import random
+import json
+
+
+# API_URL = 'http://0.0.0.0:5000'
+API_URL = 'http://127.0.0.1:5001'
+
+
+# -------------- Fonctions pour calculer les stats sur les syllabes ------------------------- #
+
+
+# Ici, pour un mot donné, on incrémente les statistiques de ses syllabes adjacentes
+def get_statistics(language,word):
+    i = 0
+    length = len(word['syllables_ipa'])
+    for syllable in word['syllables_ipa']:
+
+        # récupération de la syllabe dans la BD ou création d'une nouvelle
+        syll = requests.get(f'{API_URL}/{language}_syllables/{syllable}').json()['result']
+        new = False
+        if type(syll) is str:
+            new = True
+            try :
+                ortho = word['syllables'][i]
+            except (KeyError,IndexError):
+                ortho = ""
+            syll = {
+                'language' : language,
+                'ipa_syllable' : syllable,
+                'orthographical_syllable' : ortho,
+                'preceding_ipa_syllable' : {},
+                'following_ipa_syllable' : {}
+            }
+
+        # ajout de la syllabe précédente dans les statistiques
+        if i > 0:
+            previous = word[i-1]
+        else:
+            previous = " "
+        try:
+            syll['preceding_ipa_syllable'][previous] += 1
+        except KeyError:
+            syll['preceding_ipa_syllable'][previous] = 1
+        except TypeError:
+            syll['preceding_ipa_syllable'] = {previous : 1}
+
+        # ajout de la syllabe suivante dans les statistiques
+        if i < length - 1:
+            next = word[i+1]
+        else:
+            next = " "
+        try:
+            syll['following_ipa_syllable'][next] += 1
+        except KeyError:
+            syll['following_ipa_syllable'][next] = 1
```

Comment:
maybe you can use a defaultdict instead, in this case a defaultdict of defaultdict: `syll = defaultdict(lambda: defaultdict(int))` this means that if the key is not present, it does not fail but assume the default value of int (0).